### PR TITLE
PCHR-3403: Workflow configuration fixes

### PR DESCRIPTION
**Pull request by reneolivo**
This PR fixes the following issues in the Workflow configuration screen:

* Removes an extra border that was being displayed between the UI tabs panel and the timeline table
* Displays the 2px border radius of the workflow configuration container.

### Before
new workflow type hr17 9000 11

### After
new workflow type hr17 9000 10

### Technical details

The `scss/civicrm/administer/civi-case/_case-types.scss` page which corresponds to the workflow configuration was fixed with the following changes:

* Fixed lint issues present in the file ([see changes here](https://github.com/civicrm/org.civicrm.shoreditch/pull/176/commits/53fc86ff58875c0126ff6f11aa735a8cf29a7f93)).
* Removed the border top of the table inside the UI panel:
```scss
.ui-tabs-panel {
  padding: 0 !important;

  div[ng-include] table {
    border-top: 0; // here
  }
}
```
*  Removed the background for `.crm-submit-buttons` to allow the borders radius of their parent `.crm-container` to be displayed:
```scss
.crm-submit-buttons {
  background: none !important; // here
  border-top: 0;
}
```

### CODE REVIEW

File: scss/civicrm/administer/civi-case/_case-types.scss [hunk 1]
```diff
@@ -34,6 +35,10 @@
 
       .ui-tabs-panel {
         padding: 0 !important;
+
+        div[ng-include] table {
```
Comment (AkA84) [inline scss/civicrm/administer/civi-case/_case-types.scss]: This selector looks very suspicious, using an angular attribute and two generic tags. There must be a better way.

File: scss/civicrm/administer/civi-case/_case-types.scss [hunk 2]
```diff
@@ -44,7 +49,8 @@
 
   #crm-main-content-wrapper {
     .crm-submit-buttons {
-      border-top: none;
+      background: none !important;
```
Comment (AkA84) [inline scss/civicrm/administer/civi-case/_case-types.scss:52]: what's this for? I don't see any mention of button background in the PR or in the screenshot

**Review by AkA84 — CHANGES_REQUESTED** [review]
@reneolivo where do you introduce the 2px border? I can't find it in the diffs. And why was it missing in the first place?

**Comment by reneolivo** [comment]
> @reneolivo where do you introduce the 2px border? I can't find it in the diffs. And why was it missing in the first place?

@AkA84 the 2px border is part of Shoreditch for `.crm-form-block` elements: 
https://github.com/compucorp/org.civicrm.shoreditch/blob/staging/scss/civicrm/common/_base.scss#L50

Probably my wording was not right, but this was not about adding the 2px border-radius, but removing a background that was hiding it.

**Comment by AkA84** [comment]
ok @reneolivo so then why was the background hiding the border? Is it a regression issue?

**Comment by reneolivo** [comment]
@AkA84 this is not a regression issue as far as I have searched. The problem is how the border radius is applied elsewhere because it seems over-complicated:

`.crm-form-block` is the one that usually has a white background and border radius of 2px ([defined here](https://github.com/compucorp/org.civicrm.shoreditch/blob/staging/scss/civicrm/common/_base.scss#L50))

And then, depending on the page, you have `.crm-submit-buttons` with a border radius for the top one and another for the bottom one. An example for this is the page for data customization:

```scss
&#{civi-page('admin-custom-group')} {
  // ...
  #Group {
    .crm-form-block {
      background: none; // removes background and makes the radius invisible as well
      box-shadow: none;
    }
  }

  // last element on this page has a white background with border radius at the bottom:
  #field_page + .action-link {
    background: $crm-white !important;
    border-radius: 0 0 $border-radius-base $border-radius-base;
    // ...
  }
  // ...
}

&#{civi-page('admin-custom')} {
  // ...
  #crm-main-content-wrapper {
    // first element has background and border radius top:
    .help + .crm-submit-buttons,
    .crm-submit-buttons:first-child {
      border-bottom: 1px solid $gray-lighter;
      border-radius: 2px 2px 0 0;
      border-top: 0;
    }

    // last element has background and border radius bottom:
    .crm-submit-buttons {
      background-color: $crm-white !important;
      border-radius: 0 0 2px 2px;
    }
  }
  // ..
}

/*
  Things in between have a white background because their are wrapped inside
  .crm-content-block elements. The structure is something like this:

  .crm-form-block <--- remove background
    .help
    .crm-submit-buttons <--- add border radius top and white background
    table.form-layout or .crm-content-block <--- has white background
    .crm-submit-buttons <---- add border radius bottom and white background
*/
```

Sample Source:
https://github.com/compucorp/org.civicrm.shoreditch/blob/staging/scss/civicrm/administer/customize-data/_customize-data.scss#L134

Sample pages:
* /civicrm/admin/custom/group/?action=browse
* /civicrm/admin/custom/group?action=add&reset=1

And pages where `.crm-submit-buttons` covers the border radius of their parent `.crm-form-block` element just like in this case:
* /civicrm/contact/add?reset=1&ct=Individual

So for this page, instead of adding border radius to the `.crm-submit-buttons` elements this PR removes their background and allows `.crm-form-block` to display its borders properly.

**Comment by reneolivo** [comment]
@AkA84 I have changed the selector to this one:

```scss
.ui-tabs-panel {
  // ...

  table:first-child {
    border-top: 0;
  }
}
```

Because we are only interested in removing the border top from the first `table` inside `.ui-tabs-panel` elements:

```pug
.ui-tabs-panel
  div[ng-include]
    table // we want to remove the border top from this one
    table // not from this other one
```

There are 2 considerations here:

1. Right now there's only one table in the template, I could remove the top border for all of them and it will remove just the one. I'm just trying to be more specific.
2. I'm aware that `table:first-child` would select any table that is the first child of any element in between, so for our previous example this could happen:
```pug
.ui-tabs-panel
  div[ng-include]
    table // would be selected
    div
      table // would be selected as well (not good)
```
The alternative would be to use `.ui-tabs-panel > div > table:first-child` but then that leaves the question about "what is this div element in the middle of this selector?" the only way to tell apart this div is by its ng-include attribute

**Review by AkA84 — APPROVED** [review]

**Pull request merged**